### DISCUSSION TITLE
Is there an example how to handle (no pun intended) shared_ptr like opencascade::handle ?

### DISCUSSION THREAD

**Discussion by nyholku** [bytedeco/javacpp#536] [Q&A] [answered]
OpenCascade uses its own `opencascade::handle` to refer to its objects in the function calls and return values.

This has close to 100% compatible interface (functions and operators) with `std::shared_ptr`.

Is there some example code or project which I could have a look into to see how this is should handled with JavaCPP?

I've read 

https://github.com/bytedeco/javacpp/wiki/Mapping-Recipes#using-adapters-for-c-container-types

but have not yet really understood all of it, but it will come I'm sure.

cheers Kusti

**Comment by saudet** [comment]
The "sample projects" are all here: https://github.com/bytedeco/javacpp-presets

**Comment by nyholku** [comment, marked as answer]
Thanks, I know. 

Any specific project that would implement its own 'shared pointer' that I could start with?

I know, noobs are a nuisance, but we all have to start somewhere and I've been reading all I can (including everything you've linked to so far) for the past 24 hours before I came here with my questions :)

**Reply by saudet** [reply]
The ones that show up with a search for SHARED_PTR_NAMESPACE:
https://github.com/bytedeco/javacpp-presets/search?q=SHARED_PTR_NAMESPACE

**Reply by saudet** [reply]
Oh, you mean it's not like something::shared_ptr? You could try to modify the one for shared_ptr here and adapt it to your needs:
https://github.com/bytedeco/javacpp/blob/master/src/main/java/org/bytedeco/javacpp/tools/Generator.java#L1274-L1305

**Reply by saudet** [reply]
OpenCV has its own smart pointer, but it's not exactly like std::shared_ptr. Anyway, its adapter is available here for reference:
https://github.com/bytedeco/javacpp-presets/blob/master/opencv/src/main/resources/org/bytedeco/opencv/include/opencv_adapters.h

**Reply by nyholku** [reply]
Thanks ! Those looks like something I can start with.
